interpreter/cling/test/Plugins/Simple.C

```diff
@@ -0,0 +1,50 @@
+// RUN: %rm %T/DemoPlugin%shlibext
+// RUN: clang -fno-rtti -std=c++11 %plugin_compilation_opts -shared -DDEMO_PLUGIN %s -o%T/DemoPlugin%shlibext
+// RUN: cat %s | %cling -fplugin=%T/DemoPlugin%shlibext | FileCheck %s
+
+#ifdef DEMO_PLUGIN
+
+#include "clang/Frontend/FrontendPluginRegistry.h"
+struct PluginConsumer : public clang::ASTConsumer {
+  virtual bool HandleTopLevelDecl(clang::DeclGroupRef DGR) {
+    llvm::outs() << "PluginConsumer::HandleTopLevelDecl\n";
+    return true; // Happiness
+  }
+};
+
+template<typename ConsumerType>
+class Action : public clang::PluginASTAction {
+protected:
+  std::unique_ptr<clang::ASTConsumer>
+  CreateASTConsumer(clang::CompilerInstance&, llvm::StringRef /*InFile*/) override {
+    llvm::outs() << "Action::CreateASTConsumer\n";
+    return std::unique_ptr<clang::ASTConsumer>(new ConsumerType());
+  }
+
+  bool ParseArgs(const clang::CompilerInstance &,
+                 const std::vector<std::string>&) override {
+    llvm::outs() << "Action::ParseArgs\n";
+    // return false; // Tells clang not to create the plugin.
+    return true; // Happiness
+  }
+
+  PluginASTAction::ActionType getActionType() override {
+    return AddBeforeMainAction;
+  }
+};
+
+// Register the PluginASTAction in the registry.
+static clang::FrontendPluginRegistry::Add<Action<PluginConsumer> >
```

Comment:
Can you add that to libCling, just as we have `cling::test::TestProxy` in cling/lib/Interpreter/InterpreterCallbacks.cpp?

Reply:
If I do this I will not test [what I want](https://github.com/root-project/root/pull/2371/files#diff-5d5c7607298249e0bb8b046d9e2ef6acR198). All other comments should be addressed.

Reply:
Ah, I see what you mean. The upside of the current approach is that users can see how to create a clang cling plugin (whereas sinking it in InterpreterCallbacks will make it more obscure).

Reply:
During the next decade, we can still point the two interested persons to that source file ;-) I'm favoring reduced test time and complexity.

Reply:
The comment should be addressed.